zerver/tests/test_signup.py

```diff
@@ -1448,16 +1452,20 @@ def test_confirmation_key_of_wrong_type(self) -> None:
         self.assertEqual(cm.exception.error_type, ConfirmationKeyException.DOES_NOT_EXIST)
 
         # Verify that using the wrong type doesn't work in the main confirm code path
-        email_change_url = create_confirmation_link(user, 'host', Confirmation.EMAIL_CHANGE)
+        email_change_url = create_confirmation_link(prereg_user, 'host', Confirmation.EMAIL_CHANGE)
         email_change_key = email_change_url.split('/')[-1]
         url = '/accounts/do_confirm/' + email_change_key
         result = self.client_get(url)
         self.assert_in_success_response(["Whoops. We couldn't find your "
                                          "confirmation link in the system."], result)
 
     def test_confirmation_expired(self) -> None:
-        user = self.example_user('hamlet')
-        url = create_confirmation_link(user, 'host', Confirmation.USER_REGISTRATION)
+        email = self.nonreg_email("alice")
+        realm = get_realm('zulip')
+        inviter = UserProfile.objects.first()
```

Comment:
The `.first()` thing got forgotten here :thinking:

Reply:
ah! sorry! I've changed but it got on the wrong PR so I undid the changes and forgot to add again. I'm fixing now.